src/main/fc/fc_msp.c

```diff
@@ -748,16 +748,30 @@ static bool mspCommonProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst, mspPostProce
         sbufWriteU8(dst, batteryConfig()->currentMeterSource);
         break;
 
-    case MSP_TRANSPONDER_CONFIG:
+    case MSP_TRANSPONDER_CONFIG: {
 #ifdef TRANSPONDER
-        sbufWriteU8(dst, 1); //Transponder supported
-        for (unsigned int i = 0; i < sizeof(transponderConfig()->data); i++) {
-            sbufWriteU8(dst, transponderConfig()->data[i]);
+        sbufWriteU8(dst, TRANSPONDER_PROVIDER_COUNT);
```

Comment:
Okay, I'll remove the backward compatibility changes (from local repo) insert non-backward compatibility remarks as comments.